### PR TITLE
180 bioccheck

### PR DESCRIPTION
**Pull request by arkadiuszbeer**
Task https://github.com/insightsengineering/idr-tasks/issues/180

As expected:
warnings https://github.com/insightsengineering/helios/runs/4507048112?check_suite_focus=true
error https://github.com/insightsengineering/helios/runs/4507048112?check_suite_focus=true
OK https://github.com/insightsengineering/helios/runs/4507550644?check_suite_focus=true

### CODE REVIEW

File: action.yaml [hunk 1]
```diff
@@ -120,6 +120,11 @@ inputs:
     description: "version of cache. To clean cache bump this version."
     required: false
     default: "cache-v1"
+  allow_failure:
+    description: BiocCheck errors will not fail,but will give a warning
```
Comment (knightdave) [inline action.yaml]:
```suggestion
    description: BiocCheck errors will not fail, but will give a warning.
```

**Comment by knightdave** [comment]
Add description of new parameter to `README.md`

File: action.yaml [hunk 2]
```diff
@@ -219,9 +224,11 @@ runs:
 
       - name: Check bioccheck status
         run: |
-          if [[ -f ".bioccheck_fail" ]]; then
+          if [[ -f ".bioccheck_fail"  &&  "${{ inputs.allow_failure }}" == "false" ]]; then
             echo "❌ BiocCheck has errors"
             exit 1
+          elif  [[ -f ".bioccheck_fail" ]]; then
+             echo "⚠️ BiocCheck status WARNING"
```
Comment (cicdguy) [inline action.yaml]:
```suggestion
             echo "⚠️ BiocCheck produced WARNINGs"
```

File: action.yaml [hunk 3]
```diff
@@ -120,6 +120,11 @@ inputs:
     description: "version of cache. To clean cache bump this version."
     required: false
     default: "cache-v1"
+  allow_failure:
```
Comment (cicdguy) [inline action.yaml]:
```suggestion
  allow-failure:
```

File: action.yaml [hunk 4]
```diff
@@ -219,9 +224,11 @@ runs:
 
       - name: Check bioccheck status
         run: |
-          if [[ -f ".bioccheck_fail" ]]; then
+          if [[ -f ".bioccheck_fail"  &&  "${{ inputs.allow_failure }}" == "false" ]]; then
```
Comment (cicdguy) [inline action.yaml]:
```suggestion
          if [[ -f ".bioccheck_fail"  &&  "${{ inputs.allow-failure }}" == "false" ]]; then
```

File: action.yaml (hunk 3 above)
Comment (arkadiuszbeer) [reply action.yaml]: done

**Comment by arkadiuszbeer** [comment]
> Add description of new parameter to `README.md`

Done

**Review by cicdguy — APPROVED** [review]

**Review by knightdave — APPROVED** [review]

**Pull request merged**